### ISSUE TITLE
Better AWS Taging strategy

### ISSUE THREAD

**Issue by Cryowatt** [quintilesims/layer0#139]
I would like to use a Resource Group in AWS to view and manage my AWS resources. However, L0 does not have a robust enough tagging to be able to properly find my resources. For example, if search for tags with name like 'l0-prefix' then I miss all the load balancers as they are just named, and aren't actually tagged. Ideally every AWS resource should be tagged with the L0 environment name, job, service and deploy IDs (as applicable to each resource).

**Comment by diemonster** [comment]
I agree that more robust tagging would be, overall, a beneficial thing to have. However, Layer0 uses `terraform` to view and manage AWS resources being used for a given Layer0 instance. This is available today and useable if you are blocked on trying to view/manage resources.

The disadvantage of Terraform is that it's easy to get in a weird state w resources if you're doing some things in the AWS UI/CLI and some things with terraform. Terraform state is also local to your machine rather than "on AWS" so there is no locking mechanisms, etc.

Referencing https://github.com/quintilesims/layer0/issues/9 as it's becoming more and more clear to me that we need to think about this part of the Layer0 toolchain sooner than later.

**Comment by zpatrick** [comment]
this also applies to resources created by the layer0 api
